src/timepicker/timepicker.ts

```diff
@@ -269,6 +273,31 @@ export class NgbTimepicker implements ControlValueAccessor,
     }
   }
 
+  _filter(e: KeyboardEvent) {
+    // tslint:disable-next-line
+    const keyCode = e.keyCode;
+    const ctrlKey = e.ctrlKey;
+    const metaKey = e.metaKey;
+    const shiftKey = e.shiftKey;
+    if ([46, 8, 9, 27, 13].indexOf(keyCode) !== -1 ||  // Allow: Delete, Backspace, Tab, Escape, Enter
+        (keyCode === 65 && ctrlKey === true) ||        // Allow: Ctrl+A
```

Comment:
not sure if this is any better but can also do something like
```typescript
([65, 67, 86, 88].indexOf(keyCode) !== -1  && (ctrlKey === true || metaKey === true)) ||    // Allow: Ctrl+A
```

Reply:
Thanks @msosa, yes you're right. I took this code somewhere on internet, but never had a closer look at the "if" part (it was the more complete I found, taking account of such cases). I'm going to change it.